aws/ec2/resources.py

```diff
@@ -30,3 +29,25 @@ def ec2_ebs_volumes():
       .extract_key('Volumes')\
       .flatten()\
       .values()
+
+
+def ec2_security_groups_with_in_use_flag():
+    """Returns security groups with an additional "InUse" key,
+    which is True if it is associated with at least one EC2
+    instance.
+    """
+    sec_groups = ec2_security_groups()
+    instances = ec2_instances()
+
+    in_use_sec_group_ids = defaultdict(int)
+    for instance in instances:
+        for attached_sec_group in instance['SecurityGroups']:
+            in_use_sec_group_ids[attached_sec_group['GroupId']] += 1
+
+    for sec_group in sec_groups:
+        if sec_group["GroupId"] in in_use_sec_group_ids.keys():
+            sec_group["InUse"] = True
+        else:
+            sec_group["InUse"] = False
```

Comment:
This is fine for now, but my personal preference is to not modify the resources to make the tests easier to understand without digging into `resources.py`.